src/c++/perf_analyzer/request_rate_worker.cc

```diff
@@ -60,25 +60,28 @@ RequestRateWorker::Infer()
 void
 RequestRateWorker::CreateContext()
 {
-  thread_stat_->status_ = factory_->CreateClientBackend(&(ctx_.infer_backend_));
-  ctx_.options_.reset(new cb::InferOptions(parser_->ModelName()));
-  ctx_.options_->model_version_ = parser_->ModelVersion();
-  ctx_.options_->model_signature_name_ = parser_->ModelSignatureName();
+  ctxs_.emplace_back(new InferContext);
```

Comment:
nit: Can you use `std::vector::push_back()` and `std::make_shared<>()` instead of directly calling `new`?

Reply:
Is that better? I suppose either way you are pushing back a pointer instead of a full object, so the copy isn't very expensive?

Reply:
`new` is generally recommended to be avoided being called directly. If you use `std::make_shared<>()` and forget to do anything with it, it'll still be destructed, whereas the output of `new`, if forgotten, results in memory leak.

Reply:
fixed